test/index.js

```diff
@@ -13,23 +14,153 @@ const promisifyApi = (o) => {
   const close = promisify(o.close)
   return {
     __proto__: o,
-    connect, 
+    connect,
     lookupOne,
     lookup,
     bind,
     close
   }
 }
 
-const close = (nets) => Promise.all(nets.map((n) => n.close()))
+test('socket recieved – socket option – (address details based connection)', async ({is, plan}) => {
+  // there is an unfixed stateful bug which makes
+  // this test non-atomic causing the socket method to never be called. 
+  // Temporary workaround is to place this test first – TODO: move rest to above 
+  // other socket recieved test
+  // NOTE: it may be utp-native, as even stripping all modules except utp-native 
+  // out of the require cache to get a fresh state fails. We can't strip utp-native
+  // from the require cache, it leads to an error (probably to do with Node expecations
+  // around native bindings, error is Module did not self register)
+  plan(1)
+  const network = promisifyApi(guts({
+    socket(sock) {
+      is(sock instanceof Socket, true)
```

Comment:
could be a utp socket

Reply:
is it a race or env based?

Reply:
race, depends on the network setup etc. first one that goes through wins

Reply:
ok so best thing would probably be to use an || and allow either to be true